pulp_ostree/tests/functional/api/test_modify.py

```diff
@@ -1,248 +1,321 @@
-        distribution_data = OstreeOstreeDistribution(
-            **gen_distribution(repository=self.repo2.pulp_href)
-        )
-        response = self.distributions_api.create(distribution_data)
-        distribution = monitor_task(response.task).created_resources[0]
-        self.addCleanup(self.distributions_api.delete, distribution)
-
-        ostree_repo_path = self.distributions_api.read(distribution).base_url
-
-        remote_name = init_local_repo_with_remote(self.remote.name, ostree_repo_path)
-        self.addCleanup(shutil.rmtree, self.remote.name)
-        validate_repo_integrity(
-            self.remote.name,
-            f"{remote_name}:{ref.name}",
-            commits_to_check={latest_commit.checksum},
-            depth=0,
+@pytest.fixture
+def synced_repo_version(
+    ostree_repositories_api_client,
+    ostree_repositories_versions_api_client,
+    ostree_repository_factory,
+    ostree_remote_factory,
+    monitor_task,
+):
+    """
+    Fixture that syncs a Remote ostree Repository,
+    waits until the sync task completes and returns the
+    created repository version object.
+    """
+    repo = ostree_repository_factory()
+    remote = ostree_remote_factory(url=OSTREE_FIXTURE_URL, policy="immediate")
+    result = ostree_repositories_api_client.sync(repo.pulp_href, {"remote": remote.pulp_href})
+    repo_version_href = monitor_task(result.task).created_resources[0]
+    return ostree_repositories_versions_api_client.read(repo_version_href), remote, repo
+
+
+@pytest.fixture
+def modify_repo(bindings_cfg, monitor_task):
+    """Modify a repository.
+
+    :param repo: A dict of information about the repository.
+    :param base_version: If provided, use a specific repository version
+        instead of the latest one.
+    :param add_units: A list of dicts of information about the content
+        units to add.
+    :param remove_units: A list of dicts of information about the content
+        units to remove.
+    :returns: The server's response. A dict of information about the just
+        created modification operation.
+    """
+
+    def _modify_repo(repo, base_version=None, add_units=None, remove_units=None):
+        params = {}
+        if add_units:
+            params["add_content_units"] = [content["pulp_href"] for content in add_units]
+        if remove_units:
+            if remove_units == ["*"]:
+                params["remove_content_units"] = remove_units
+            else:
+                params["remove_content_units"] = [content["pulp_href"] for content in remove_units]
+        if base_version:
+            params["base_version"] = base_version
+
+        response = requests.post(
+            url=urljoin(bindings_cfg.host, repo["pulp_href"] + "modify/"),
+            data=params,
+            auth=HTTPBasicAuth(bindings_cfg.username, bindings_cfg.password),
         )
+        monitor_task(response.json()["task"])
+
+    return _modify_repo
+
+
+@pytest.fixture
+def get_content(bindings_cfg):
```

Comment:
I believe you can delete this.

In the past, we replaced `get_content` from `pulp_smash` by a plain call to `content_api.list(repo.latest_version)`, see https://github.com/pulp/pulp_file/commit/5c65d9238eed703b31c453f64bf4df4adedfa492#diff-082889e4107e892cad9e326317979e0a6e6b82a021fb85db8228af551797d6a0R99. Considering this, I propose adopting a similar strategy in the current context.

Furthermore, we will need to modify or remove the `normalize_content` function to parse results separately for each content type. Alternatively, we can make the function a fixture and fuse the `content_api.list(repo.latest_version)` logic into it, so we will have the downloading and normalizing at one place. What do you think?

Reply:
hum... not sure if I understood it, but both options seem complicated. I'll do some tests and revisit/re-read the functions to have a better understanding of the proposals.

Reply:
The basic idea is to use bindings to get things done. We should stray away from using `requests` to access API endpoints that are suited for administering. In this case, we want to get a list of content units from an API endpoint which is properly mapped by the bindings.